### PR TITLE
[BC Break] Retire redux

### PR DESCRIPTION
**Pull request by fzaninotto**
- [x] Remove `<Admin customReducers>`
- [x] Remove `<Admin initialState>`
- [x] Do not create a Redux `<Provider>` in `<Admin>`
- [x] Remove dependencies on `redux` and `react-redux` packages
- [x] Make sure tests and demos still work
- [x] Fix doc
- [x] Add changelog

### CODE REVIEW

**Comment by fzaninotto** [comment]
I'm surprised that the build of the e-commerce demo doesn't change much:

```
File sizes after gzip:

  607.9 kB (-1.69 kB)  build/static/js/main.886b6121.js
  85.79 kB             build/static/js/327.3776b9a6.chunk.js
  67.96 kB             build/static/js/775.729f778c.chunk.js
  58.95 kB             build/static/js/250.42aa2365.chunk.js
  12.43 kB             build/static/js/292.eea62de7.chunk.js
  5.73 kB (+1 B)       build/static/js/908.0296630d.chunk.js
  3.95 kB              build/static/js/376.72118f7b.chunk.js
  3.73 kB              build/static/js/477.84064d46.chunk.js
  3.61 kB              build/static/js/247.4f700bc0.chunk.js
  1.43 kB (+1 B)       build/static/js/151.cf8ab998.chunk.js
```

While bundlephobia tells me that [react-redux alone weighs 5,4 kB](https://bundlephobia.com/package/react-redux@7.2.6).

**Review by WiXSL — COMMENTED** [review]
There is still a dependency to `"@types/react-redux": "^7.1.1"`, in the main `react-admin/package.json`, I don't know if it's necessary

File: UPGRADE.md [hunk 1]
```diff
@@ -1,5 +1,335 @@
...
+        >
+            ...
+        </Admin>
++   </Provider>
+);
+```
+
+### `<Admin initialState>` Is Gone
+
+The `<Admin initialState>` prop used to be used to initialize the store. As there is no store, you cannot set the react-admin defaults this way.
```
Comment (djhi) [inline UPGRADE.md:64]:
```suggestion
The `<Admin initialState>` prop used to be used to initialize the redux store. As there is no redux store anymore, you cannot set the react-admin defaults this way.
```

File: UPGRADE.md [hunk 2]
```diff
@@ -1,5 +1,335 @@
...
+### Action Creators Are Gone
+
+As React-admin no longer uses Redux, each time your code used react-redux' `dispatch` with an action creator, you'll have to replace it with a hook. 
+
+- `dispatch(fetchStart())` and `dispatch(fetchEnd())` must be replaced by `useQuery()` and `useMutation()`
+- `dispatch(setSidebarVisibility(true))` must be replaced by `useSidebarState()`
+- `dispatch(showNotification('Comment approved', 'success'))` must be replaced by `useNotify()`
+- `dispatch(push('/comments'))` must be replaced by `useNavigate()`
+
+React-admin used `dispatch` in many other places, but they were already behind a hook (`useRecordSelection`, `useListParams`, `useExpanded`, `useNotify`, `useSidebarState`), and not documented. If you dispatched react-admin actions manually, you'll have to look for the hook alternative.
```
Comment (djhi) [inline UPGRADE.md:245]:
```suggestion
React-admin used `dispatch` in many other places, but they were already behind a hook (`useRecordSelection`, `useListParams`, `useExpanded`, `useNotify`, `useSidebarState`), and not documented. If you dispatched react-admin actions manually, you'll have to look for the hook alternatives.
```

**Review by djhi — CHANGES_REQUESTED** [review]
There's an eslint warning left

**Review by djhi — APPROVED** [review]

**Pull request merged**